docs/proposals/20230407-managed-k8s-capi-contract-changes.md

```diff
@@ -0,0 +1,556 @@
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**  *generated with [DocToc](https://github.com/thlorenz/doctoc)*
+
+- [Contract Changes to Support Managed Kubernetes](#contract-changes-to-support-managed-kubernetes)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+    - [Future Work](#future-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+      - [Story 3](#story-3)
+      - [Story 4](#story-4)
+      - [Story 5](#story-5)
+      - [Story 6](#story-6)
+      - [Story 7](#story-7)
+    - [Design](#design)
+      - [Core Cluster API changes](#core-cluster-api-changes)
+      - [Infra Providers API changes](#infra-providers-api-changes)
+      - [Core Cluster API Controllers changes](#core-cluster-api-controllers-changes)
+      - [Provider controller changes](#provider-controller-changes)
+    - [Guidelines for infra providers implementation](#guidelines-for-infra-providers-implementation)
+  - [Background work](#background-work)
+    - [EKS in CAPA](#eks-in-capa)
+      - [AKS in CAPZ](#aks-in-capz)
+      - [GKE in CAPG](#gke-in-capg)
+      - [Learnings from original Proposal: Two kinds with a Managed Control Plane & Managed Infra Cluster adhering to the current CAPI contracts](#learnings-from-original-proposal-two-kinds-with-a-managed-control-plane--managed-infra-cluster-adhering-to-the-current-capi-contracts)
+    - [Two New Flows](#two-new-flows)
+      - [Flow 1: `<Infra>Cluster` and `<Infra>ControlPlane`, with `ControlPlaneEndpoint` reported via `<Infra>ControlPlane`](#flow-1-infracluster-and-infracontrolplane-with-controlplaneendpoint-reported-via-infracontrolplane)
+      - [Flow 2: Change CAPI to make `<Infra>Cluster` optional](#flow-2-change-capi-to-make-infracluster-optional)
+      - [Alternative Option: Introduce a new Managed Kubernetes provider type (with contract)](#alternative-option-introduce-a-new-managed-kubernetes-provider-type-with-contract)
+  - [Recommendations](#recommendations)
+  - [Implementation History](#implementation-history)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+---
+title: Contract Changes to Support Managed Kubernetes
+authors:
+  - "@jackfrancis"
+reviewers:
+  - "@richardcase"
+  - "@pydctw"
+  - "@mtougeron"
+  - "@CecileRobertMichon"
+  - "@fabriziopandini"
+  - "@sbueringer"
+  - "@killianmuldoon"
+  - "@mboersma"
+  - "@nojnhuh"
+creation-date: 2023-04-07
+last-updated: 2023-04-07
+status: provisional
+see-also:
+  - "/docs/proposals/20220725-managed-kubernetes.md"
+---
+
```

Comment:
Opening a new conversation so it doesn't get lost.

I think the CAEP title should match the PR title and vice verca

xref: https://github.com/kubernetes-sigs/cluster-api/pull/8500/files#r1372011564

Reply:
done

Reply:
@jackfrancis We also have 4 occurences of the old title "Contract Changes to Support Managed Kubernetes" in this doc.

Btw, just noticed that we have 2 table of contents :)

Reply:
ya, we were not utilizing doctoc correctly, fixed